casadm/cas_lib.c

```diff
@@ -2779,7 +2777,14 @@ int list_caches(unsigned int list_format, bool by_id_path)
 			snprintf(mode_string, sizeof(mode_string), "wb->%s",
 					cache_mode_to_name(curr_cache->mode));
 		} else {
-			tmp_status = get_cache_state_name(curr_cache->state);
+			if (curr_cache->standby_detached) {
+				snprintf(status_buf, sizeof(status_buf), 
+					"Standby detached");
+				tmp_status = status_buf;
+			} else {
+				tmp_status = get_cache_state_name(curr_cache->state);
+			}
+
```

Comment:
how about adding a "detached" parameter to get_cache_state_name() and moving this entire logic to get_cache_state_name() ?